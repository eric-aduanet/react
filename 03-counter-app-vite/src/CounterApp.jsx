import { useState } from "react";
import PropTypes from "prop-types";

import "./index.css";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = (event) => {
    setCounter(counter + 1);
  };
  const handleSub = (event) => {
    setCounter(counter - 1);
  };
  const handleReset = (event) => {
    setCounter(value);
  };
  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="contador">{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSub}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
