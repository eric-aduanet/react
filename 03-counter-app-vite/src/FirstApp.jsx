import { Fragment } from "react";
import PropTypes from "prop-types";
const getSaludo = (nombre) => "Hola, " + nombre;

const newMsg = {
  msg: "Hola mundo",
  title: "Eric",
};

export const FirstApp = ({
  title = "Hola, soy Eric!",
  subtitle = "",
  name = "No hay nombre",
}) => {
  //   console.log(props);
  return (
    <Fragment>
      <h1 data-testid="test-title">{title}</h1>
      {/* <h1>{JSON.stringify(newMsg)}</h1> */}
      <h2>{subtitle}</h2>
      <p>{name}</p>
    </Fragment>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
};

import "./index.css";
