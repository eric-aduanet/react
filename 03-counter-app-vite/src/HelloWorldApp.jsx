// export function App() {
//   //document.createElement...
//   return <h1>Hola mundo!</h1>;
// }

// export const HelloWorldApp = () => {
//   return <h1>Creado con rafc</h1>;
// };

export const ArrowFunc = () => <h1>Creado con un arrow func</h1>;
