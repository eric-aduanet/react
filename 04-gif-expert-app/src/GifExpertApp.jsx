import { useState } from "react";
import "./styles.css";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  const onAddCategory = (event) => {
    // setCategories((cat) => [elem, ...cat]);
    if (categories.includes(event)) return;
    setCategories((cat) => [event, ...cat]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      {/* Input */}
      <AddCategory
        onValue={onAddCategory}
        //   setCategories={setCategories}
      />
      {categories.map((x) => (
        <GifGrid key={x} category={x} />
      ))}
    </>
  );
};
