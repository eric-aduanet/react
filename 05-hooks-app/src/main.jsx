import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import "./index.css";
// import { TodoApp } from "./08-useReducer/TodoApp";
import { MainApp } from "./09-useContext/MainApp";
import { BrowserRouter } from "react-router-dom";
// import { SimpleForm } from "./02-use-effect/SimpleForm";
// import { FormWithCustomHooks } from "./02-use-effect/FormWithCustomHook";
// import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
// import { FocusScreen } from "./04-use-ref/FocusScreen";
// import { Layout } from "./05-useLayoutEffect/Layout";
// import { Memorize } from "./06-memos/Memorize";
// import { MemorizeHook } from "./06-memos/MemorizeHook";
// import { CallbackHook } from "./06-memos/CallbackHook";
// import { Padre } from "./07-tarea-memo/Padre";
// import { CounterWithCustomHook } from "./01-use-state/CounterWithCustomHook";
// import { CounterApp } from "./01-use-state/CounterApp";
// import { HooksApp } from "./HooksApp";
// import "./08-useReducer/intro-reducer";

createRoot(document.getElementById("root")).render(
  <BrowserRouter>
    <MainApp />
  </BrowserRouter>
);
