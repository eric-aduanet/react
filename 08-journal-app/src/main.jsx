import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import "./styles.css";
import { JournalApp } from "./JournalApp.jsx";
import { AppRouter } from "./router/AppRouter.jsx";
import { BrowserRouter } from "react-router-dom";

createRoot(document.getElementById("root")).render(
  <StrictMode>
    <BrowserRouter>
      <JournalApp />
    </BrowserRouter>
  </StrictMode>
);
