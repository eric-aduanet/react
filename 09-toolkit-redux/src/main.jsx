import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import App from "./App.jsx";
import { Provider } from "react-redux";
import { store } from "./store/store.js";
import { PokemonApp } from "./PokemonApp.jsx";
import { TodoApp } from "./TodoApp.jsx";

createRoot(document.getElementById("root")).render(
  <StrictMode>
    <Provider store={store}>
      {/* <PokemonApp /> */}
      <TodoApp />
    </Provider>
  </StrictMode>
);
