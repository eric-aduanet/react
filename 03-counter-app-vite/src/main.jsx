import React from "react";
import ReactDOM from "react-dom/client";
// import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
// import { ArrowFunc } from "./HelloWorldApp";
// import { App } from "./HelloWorldApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HelloWorldApp />:w
     */}
    {/* <ArrowFunc /> */}
    <FirstApp title="Hola" subtitle="Subtitle" name="Eric" />
  </React.StrictMode>
);
