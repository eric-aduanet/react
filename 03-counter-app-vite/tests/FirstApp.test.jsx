import { getByTestId, getByText, render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("FirstApp", () => {
  test("Debe hacer match con el snapshot", () => {
    const title = "Hola, soy Eric";
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
    // expect(container).toMatchSnapshot();
    // console.log(container);
  });
  test("Debe de mostrar el title en un h1", () => {
    const title = "Hola, soy Eric";
    const { container, getByText, getByTestId } = render(
      <FirstApp title={title} />
    );
    expect(getByText(title)).toBeTruthy();
    // const h1 = container.querySelector("h1");
    // expect(h1.innerHTML).toBe(title);
    expect(getByTestId("test-title").innerHTML).toBe(title);
  });

  test("Debe mostrar el subtitulo enviado por props", () => {
    const title = "Hola, soy Eric";
    const subtitle = "Hoy soy un subtitulo";
    const { container, getAllByText } = render(
      <FirstApp title={title} subtitle={subtitle} />
    );
    expect(getAllByText(subtitle).length).toBeGreaterThan(0);
  });
});
