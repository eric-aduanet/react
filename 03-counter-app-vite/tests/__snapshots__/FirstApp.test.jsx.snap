// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Eric
  </h1>
  <h2 />
  <p>
    No hay nombre
  </p>
</div>
`;
